### PR TITLE
Ps/pm 2910/browser header component

### PR DESCRIPTION
**Pull request by MGibson1**
Header component has a slots field with a left, center, right, and a right-most location for a current account, which will link to an account switcher.

## Type of change

```
- [ ] Bug fix
- [x] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Iterative work to suppose browser account switching. Designs for which include a top-right current profile avatar (component to come in separate PR). This component should be visible basically always, unless we're on a login screen and no other accounts are known.

To facilitate this design, this component provides the same functionality as the current header pattern, with slots for left, center, and right header slots, but includes the current account component should it be available.

I included it's use in a few locations, but it will be required in more as we roll out the feature.

build-feature flagged since it would be UB to have two accounts with servers with conflicting configs enabling the feature.

## Code changes

- **gitignore**: add local config to browser configs
- **lock.component/vault-filter.component**: show usage of header component.
- **flags/base.config**: define flag
- **header.component**: wraps previous `header` into a `header-content` and uses content slots to pick content as previously done through css. Displays current account placeholder unless no authed accounts exist
- **base.scss**: `header-content` needs to be flex to allow slots to fill as before.

## Screenshots

**Flag off**
image

**Flag on and accounts$ emitting authed accounts (including the one being unlocked here)**
image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
  -  I'd love to, but we don't have storybook for browser 😭 
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by MGibson1** [comment]
you know... not what I look at it, I'm not happy with this being a center that isn't centered. I'm going to make the center actually in the middle...

**Comment by MGibson1** [comment]
That's better

**Review by LRNcardozoWDF — APPROVED** [review]
Looks good to me!

**Review by andrebispo5 — DISMISSED** [review]
LGTM 👍

**Review by LRNcardozoWDF — DISMISSED** [review]

**Review by andrebispo5 — APPROVED** [review]

**Review by justindbaur — DISMISSED** [review]

**Review by LRNcardozoWDF — DISMISSED** [review]

File: apps/browser/src/platform/popup/header.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<header [ngClass]="{ 'no-theme': noTheme }">
+  <div class="header-content">
+    <ng-content select=".sr-only"></ng-content>
+    <ng-content select=".left"></ng-content>
+    <ng-content select=".center"></ng-content>
+    <div class="right">
+      <ng-content select=".right"></ng-content>
+      <ng-container *ngIf="authedAccounts$ | async">
+        TODO: Current Account
```
Comment (djsmith85) [inline apps/browser/src/platform/popup/header.component.html:9]: This should likely be a comment. Fine during dev, but we likely want to remove/hide it before the merge.

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/browser/src/platform/popup/header.component.html (hunk 1 above)
Comment (MGibson1) [reply apps/browser/src/platform/popup/header.component.html:9]: This is behind a compile time flag. It's totally inaccessible to production builds, and useful while we develop this in place

Comment (djsmith85) [reply apps/browser/src/platform/popup/header.component.html:9]: That's true, fine with me!

**Review by andrebispo5 — APPROVED** [review]

**Review by justindbaur — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Pull request merged**